File: server/java/src/main/java/com/hrms/business/abstracts/CandidateLinkService.java
package com.hrms.business.abstracts;

import com.hrms.core.utilities.results.DataResult;
import com.hrms.core.utilities.results.Result;
import com.hrms.entities.concretes.CandidateLink;

import java.util.List;

public interface CandidateLinkService {
    Result add(CandidateLink candidateLink);
    DataResult<List<CandidateLink>> getAll();
    DataResult<List<CandidateLink>> getAllByCandidateId(int candidateId);
}

File: client/react/hrms/src/services/candidateLinkService.js
import axios from "axios";

export default class CandidateLinkService {
  getAll() {
    return axios.get("http://localhost:8080/api/candidatelinks/getall");
  }

  getAllByCandidateId(candidateId) {
    return axios.get(
      "http://localhost:8080/api/candidatelinks/getallbycandidateid?candidateId=" +
        candidateId
    );
  }

  save(candidateLink) {
    return axios.post(
      "http://localhost:8080/api/candidatelinks/save",
      candidateLink
    );
  }

  delete(candidateLink) {
    return axios.post(
      "http://localhost:8080/api/candidatelinks/delete",
      candidateLink
    );
  }
}
